src/Stryker.Core/Stryker.Core/Reporters/GitBaselineReporter.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Stryker.Core.Baseline;
+using Stryker.Core.DashboardCompare;
+using Stryker.Core.Mutants;
+using Stryker.Core.Options;
+using Stryker.Core.ProjectComponents;
+using Stryker.Core.Reporters.Json;
+using System.Collections.Generic;
+
+namespace Stryker.Core.Reporters
+{
+    public class GitBaselineReporter : IReporter
+    {
+        private readonly StrykerOptions _options;
+        private readonly IBaselineProvider _baselineProvider;
+        private readonly IGitInfoProvider _gitInfoProvider;
+
+        public GitBaselineReporter(StrykerOptions options, IBaselineProvider baselineProvider = null, IGitInfoProvider gitInfoProvider = null)
+        {
+            _options = options;
+            _baselineProvider = baselineProvider ?? BaselineProviderFactory.Create(options);
+            _gitInfoProvider = gitInfoProvider ?? new GitInfoProvider(options);
+        }
+        public void OnAllMutantsTested(IReadOnlyInputComponent reportComponent)
+        {
+            var mutationReport = JsonReport.Build(_options, reportComponent);
+            var projectVersion = _gitInfoProvider.GetCurrentBranchName() ?? _options.FallbackVersion;
```

Comment:
Baselines are loaded using this location `$"dashboard-compare/{branchName}"`

This means in the best case scenario you can the fallback version every test run and in the worst case scenario (no fallback) this feature never actually works.